File: DSA/AnshulJain_590014934/Day_80_Question_3.py
class Node:
    def __init__(self,val):
        self.val=val
        self.left=None
        self.right=None
def insert(root,val):
    if not root:
        return Node(val)
    if val<root.val:
        root.left=insert(root.left,val)
    else:
        root.right=insert(root.right,val)
    return root
def kthsmallest(root,k):
    stack=[]
    while True:
        while root:
            stack.append(root)
            root=root.left
        root=stack.pop()
        k-=1
        if k==0:
            return root.val
        root=root.right

n=list(map(int,input("Enter tree nodes: ").split()))
root=None
for x in n:
    root=insert(root,x)
k=int(input("Enter k: "))
print(kthsmallest(root,k))

File: DSA/RATNA_KHANNA_590013141/DAY_80_QUESTION_3.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *left, *right;
};

struct Node* createNode(int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->left = newNode->right = NULL;
    return newNode;
}

struct Node* insert(struct Node* root, int value) {
    if (root == NULL) return createNode(value);
    if (value < root->data) root->left = insert(root->left, value);
    else if (value > root->data) root->right = insert(root->right, value);
    return root;
}

int count = 0, result = -1;

void kthSmallestUtil(struct Node* root, int k) {
    if (root == NULL || result != -1) return;
    kthSmallestUtil(root->left, k);
    count++;
    if (count == k) {
        result = root->data;
        return;
    }
    kthSmallestUtil(root->right, k);
}

int kthSmallest(struct Node* root, int k) {
    count = 0;
    result = -1;
    kthSmallestUtil(root, k);
    return result;
}

int main() {
    int n, value, i, k;
    struct Node* root = NULL;
    printf("Enter number of nodes: ");
    scanf("%d", &n);
    printf("Enter tree values: ");
    for (i = 0; i < n; i++) {
        scanf("%d", &value);
        root = insert(root, value);
    }
    printf("Enter value of K: ");
    scanf("%d", &k);
    int ans = kthSmallest(root, k);
    if (ans != -1)
        printf("Kth smallest element: %d", ans);
    else
        printf("K is larger than number of nodes in the tree");
    return 0;
}
